### PR TITLE
Updated fetch_all_by_Transcript method to optionally filter by featur…

### PR DESCRIPTION
**Pull request by premanand17**
…e_type (dna_align or protein_align features)

### CODE REVIEW

File: modules/Bio/EnsEMBL/DBSQL/TranscriptSupportingFeatureAdaptor.pm [hunk 1]
```diff
@@ -75,14 +75,19 @@ use vars qw(@ISA);
 =cut
 
 sub fetch_all_by_Transcript {
-  my ( $self, $transcript )  = @_;
+  my ( $self, $transcript, $feature_type )  = @_;
```
Comment (nerdstrike) [inline modules/Bio/EnsEMBL/DBSQL/TranscriptSupportingFeatureAdaptor.pm]: Please add the new argument to the POD block above

File: modules/t/transcriptSupportingFeatureAdaptor.t [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+
+#Fetch all transcript supporting features
+my $supporting_features = $tsf_adaptor->fetch_all_by_Transcript($tr);
+isa_ok( $supporting_features, 'ARRAY' );
+ok(5 == scalar(@$supporting_features), "Fetched all transcript supporting features");
+
+
+#Fetch all transcript supporting features of type 'protein_align_feature'
+my $supporting_features_proteins = $tsf_adaptor->fetch_all_by_Transcript($tr, "protein_align_feature");
```
Comment (nerdstrike) [inline modules/t/transcriptSupportingFeatureAdaptor.t]: Consider adding a fail case with a funky feature type to examine the exception.

File: modules/t/transcriptSupportingFeatureAdaptor.t [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+# limitations under the License.
+
+use strict;
+use Test::More;
+use Test::Warnings;
+use Bio::EnsEMBL::Test::MultiTestDB;
+
+use Bio::EnsEMBL::DnaDnaAlignFeature;
+use Bio::EnsEMBL::Test::TestUtils;
+use Data::Dumper;
```
Comment (nerdstrike) [inline modules/t/transcriptSupportingFeatureAdaptor.t]: This is probably a relic

File: modules/Bio/EnsEMBL/DBSQL/TranscriptSupportingFeatureAdaptor.pm [hunk 4]
```diff
@@ -99,23 +104,34 @@ sub fetch_all_by_Transcript {
...
-  return $out;
+	  $sth->finish();
+
+   
+   if(defined $feature_type){
+   	return $out_feature_type->{$feature_type};
+   }else{
+   
+  	while(my ($feature_type, $new_features) = each(%$out_feature_type)){
+  		push @$out, @{$new_features};
```
Comment (nerdstrike) [inline modules/Bio/EnsEMBL/DBSQL/TranscriptSupportingFeatureAdaptor.pm]: We have a no \t policy in this repo. It looks like this is tabbed in. Preferred substitution is 4 or 2 spaces.

File: modules/Bio/EnsEMBL/DBSQL/TranscriptSupportingFeatureAdaptor.pm [hunk 5]
```diff
@@ -63,26 +63,34 @@ use vars qw(@ISA);
 
   Arg [1]    : Bio::EnsEMBL::Transcript $transcript 
                The transcript to fetch supporting features for
+  Arg [2]	 : String (optional)
```
Comment (lairdm) [inline modules/Bio/EnsEMBL/DBSQL/TranscriptSupportingFeatureAdaptor.pm]: Your editor still seems to be adding tabs, so things aren't aligning properly, please check your editor so these are spaces.

File: modules/Bio/EnsEMBL/DBSQL/TranscriptSupportingFeatureAdaptor.pm [hunk 6]
```diff
@@ -99,23 +107,33 @@ sub fetch_all_by_Transcript {
   while(my ($type, $feature_id) = $sth->fetchrow){
     if($type eq 'protein_align_feature'){
       $feature = $prot_adp->fetch_by_dbID($feature_id);
-    } elsif($type eq 'dna_align_feature'){
+     } elsif($type eq 'dna_align_feature'){
```
Comment (lairdm) [inline modules/Bio/EnsEMBL/DBSQL/TranscriptSupportingFeatureAdaptor.pm]: Pretty spacing please, a nitpick I know.

**Comment by nerdstrike** [comment]
I think this is good for merging now

**Pull request merged**